app/(viewport)/form/page.tsx

```diff
@@ -0,0 +1,181 @@
+'use client';
+
+import { useRouter } from 'next/navigation';
+
+import { useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+
+import { zodResolver } from '@hookform/resolvers/zod';
+
+import { postForm } from '@/api/postForm';
+import Step1 from '@/components/steps/Step1';
+import Step2 from '@/components/steps/Step2';
+import Step3 from '@/components/steps/Step3';
+import Step4 from '@/components/steps/Step4';
+import { Button } from '@/components/ui/button';
+import {
+  AdditionalCategories,
+  BalanceGameCategories,
+  BasicInformationCategories,
+  Category,
+  ErrorMessages,
+  SelfPromotionCategories,
+} from '@/constants/categories';
+import { StepType, Steps } from '@/constants/steps';
+import { RegisterSchema, type RegisterSchemaType } from '@/lib/registerSchema';
+
+export default function Form() {
+  const router = useRouter();
+  const methods = useForm<RegisterSchemaType>({
+    resolver: zodResolver(RegisterSchema),
+    defaultValues: {
+      mbti: '',
+      name: '',
+      gender: '',
+      email: '',
+      instagramId: '',
+      type: '',
+      animal: '',
+      situation: [
+        { scenario: 0, choice: 0 },
+        { scenario: 1, choice: 0 },
+        { scenario: 2, choice: 0 },
+        { scenario: 3, choice: 0 },
+      ],
+    },
+  });
+  const [step, setStep] = useState(0);
+
+  const getStepFields = () => {
+    switch (step) {
+      case 0:
+        return BasicInformationCategories;
+      case 1:
+        if (methods.getValues(Category.age)) return AdditionalCategories;
+        else {
+          methods.clearErrors('age');
+          return [];
+        }
+      case 2:
+        return SelfPromotionCategories;
+      case 3:
+        return BalanceGameCategories;
+      default:
+        return [];
```

Comment:
```suggestion
      default:
        throw new Error("Invalid step");
```

(p4) switch문의 default에는 에러를 발생시키는 것이 코드 퀄리티 측면에서 좋을 것 같습니다.